build.gradle.kts

```diff
@@ -1,29 +1,135 @@
+import org.gradle.api.tasks.PathSensitivity.NONE
+import org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsRootPlugin.Companion.kotlinNodeJsRootExtension
+import org.jetbrains.kotlin.gradle.targets.js.yarn.yarn
+
 plugins {
-    kotlin("jvm") version "2.1.0"
+    kotlin("multiplatform") version "2.1.0"
     kotlin("plugin.serialization") version "2.1.0"
-    application
+    id("io.kotest.multiplatform") version "5.9.1"
+    distribution
 }
 
-repositories {
-    mavenCentral()
+val collectGoodTypings by tasks.registering(Sync::class) {
+    from("../github-actions-typing-catalog/typings")
+    include("**/action-types.yml")
+    into(layout.buildDirectory.dir("good-typings"))
+    eachFile {
+        relativePath = RelativePath(
+            relativePath.isFile,
+            relativePath
+                .segments
+                .joinToString("_")
+                .replace("_action-types", "")
+        )
+    }
+    includeEmptyDirs = false
 }
```

Comment:
Do I get correctly that the bad typings are included in the resources here, and the good ones are fetched **only** from the catalog? If yes, I'm generally all for testing this action on the typings from the catalog, but I'd include test cases for the good typings independently from the catalog, for the sake of making this action self-contained.

Reply:
The "problem" is, you can easily create bad typings for each possible error.
But the good typings are not that clearly creatable and nameable imho.
Actually I think we could even write one typings file that contains all possible good cases.
My intention was to validate against all schemas in the catalog and also require that those are part of the build.
Even when adding additional good cases I would still recommend to require that the catalog typings are verified and it fails if they are not present, to make sure any change is also tested against those. 🤷‍♂️

Reply:
> Actually I think we could even write one typings file that contains all possible good cases.

Yeah, sounds good!

As I wrote, I'm totally for testing against the typings from the catalog - being able to test changes against (at least a representative subset of) the users of the library is every maintainer's dream. I even thought about listing some actions that host the typings and testing against them as well, but it's clearly more tricky and won't scale that well, so let's postpone it. The catalog is great for now.

Reply:
> I even thought about listing some actions that host the typings and testing against them as well, but it's clearly more tricky and won't scale that well, so let's postpone it. 

Actually, that's pretty trival, I also added this locally now.
The other changes I pushed already, but while doing the external typings, I used this actions typing too and found that the schema in its current state does not allow an empty typings file, so I first have to fiddle around how to fix it properly, so good I right away tried the external typings idea :-)